# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserHost.h
# Commit: 1663719e2aa5
# Full Hash: 1663719e2aa5b6f77e32398c6d6fcb4cb3c999ed
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2019-05-23 15:55:11
# Regressor Bug: 1525720
# File Overlap Count: 1
# Description:
#   Bug 1525720, part 16 - Cycle collect RemoteBrowser and support weak references in BrowserHost. r=nika
#   
#   BrowserParent is cycle collected and supported weak references, so this commit adds support
#   for these things to BrowserHost.
#   
# ==============================================================================

diff -r 53663cad8890 -r 1663719e2aa5 dom/ipc/BrowserHost.h
--- a/dom/ipc/BrowserHost.h	Wed May 08 16:12:26 2019 -0500
+++ b/dom/ipc/BrowserHost.h	Tue May 14 16:28:10 2019 -0500
@@ -32,7 +32,9 @@
  * See `dom/docs/Fission-IPC-Diagram.svg` for an overview of the DOM IPC
  * actors.
  */
-class BrowserHost : public RemoteBrowser, public nsIRemoteTab {
+class BrowserHost : public RemoteBrowser,
+                    public nsIRemoteTab,
+                    public nsSupportsWeakReference {
  public:
   typedef mozilla::layers::LayersId LayersId;
 
@@ -40,10 +42,13 @@
 
   static BrowserHost* GetFrom(nsIRemoteTab* aRemoteTab);
 
-  NS_DECL_ISUPPORTS
+  // NS_DECL_ISUPPORTS_INHERITED
+  NS_DECL_CYCLE_COLLECTING_ISUPPORTS
   // nsIRemoteTab
   NS_DECL_NSIREMOTETAB
 
+  NS_DECL_CYCLE_COLLECTION_CLASS_AMBIGUOUS(BrowserHost, RemoteBrowser)
+
   // Get the IPDL actor for the root BrowserParent. This method should be
   // avoided and consumers migrated to use this class.
   BrowserParent* GetActor() { return mRoot; }